services/worker-manager/src/provider.js

```diff
@@ -73,80 +41,45 @@ class Provider extends WMObject {
    * might wish to return {region: 'us-east-1'}
    */
   workerInfo({worker}) {
-    this._throw(errors.MethondUnimplemented, 'Provider.workerInfo()');
+    throw new Error('Method Unimplemented!');
   }
 
   /**
-   * Any code which is required to be run by this Provider must only be
-   * initiated by this method.  If there's any taskcluster-lib-iterate loops to
-   * run, this is where they should be initiated.  Once the returned promise is
-   * resolve, the Provider must be fully working.
-   *
-   * This method must not start anything which does actual provisioning, but
-   * rather is intended to do things like update pricing information or
-   * maintaining state in the underlying api.
+   * Anything a provider may want to do every provisioning loop but not tied
+   * to any one workertype. Called _before_ provision() are all called.
    */
-  async initiate() {
+  async prepare() {
   }
 
   /**
-   * Terminate any code which was started by .initiate();
+   * Given a WorkerType configuration, do whatever the provider might
+   * do with this worker type. This may mean nothing at all in the case of
+   * static provider!
```

Comment:
I think the code guarantees that this will not be called concurrently for the same workerType -- that's probably important in thinking about `providerState`, for example.

This is getting an entity from the WorkerType table, right?  Not just a workerType name?

Reply:
Yep full workerType! I guess someone could run two provisioner instances side-by-side in which case all bets are off but so long as there is only one it should be fine.